components/chat.tsx

```diff
@@ -35,34 +37,39 @@ export function Chat({ id, className, session, missingKeys }: ChatProps) {
   const [_, setNewChatId] = useLocalStorage('newChatId', id)
 
   useEffect(() => {
-    if (session?.user && !path.includes('chat') && messages.length === 1) {
-      window.history.replaceState({}, '', `/chat/${id}`);
+    if (session?.user) {
+      if (!path.includes('chat') && messages.length === 1) {
+        window.history.replaceState({}, '', `/chat/${id}`)
+      }
     }
+  }, [id, path, session?.user, messages])
 
-    const messagesLength = aiState.messages?.length;
+  useEffect(() => {
+    const messagesLength = aiState.messages?.length
     if (messagesLength === 2) {
-      router.refresh();
+      router.refresh()
     }
+  }, [aiState.messages, router])
 
-    setNewChatId(id);
-
-    missingKeys.forEach(key => {
-      toast.error(`Missing ${key} environment variable!`);
-    });
-  }, [id, path, session?.user, aiState.messages, router, setNewChatId, missingKeys]);
-
-  // send conversation start to segment if msg queue is empty
   useEffect(() => {
+    setNewChatId(id)
+    console.log('here')
     if (messages.length === 0) {
       analytics.track({
-        userId: "123",
-        event: "Conversation Started",
+        userId: '123',
+        event: 'Conversation Started',
         properties: {
           conversationId: id
         }
-      });
+      })
     }
-  }, []);
+  }, [])
```

Comment:
I reverted this file back to the original state from Vercel. We can fire the conversation started when component mounts.